dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task220829CreateExperimentsTable.java

```diff
@@ -0,0 +1,61 @@
+package com.dotmarketing.startup.runonce;
+
+
+import com.dotmarketing.common.db.DotDatabaseMetaData;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.util.Logger;
+import java.sql.SQLException;
+
+public class Task220829CreateExperimentsTable extends AbstractJDBCStartupTask {
+
+    @Override
+    public boolean forceRun() {
+        try {
+            final DotDatabaseMetaData databaseMetaData = new DotDatabaseMetaData();
+            return !databaseMetaData.tableExists(DbConnectionFactory.getConnection(), "experiment");
+        } catch (SQLException e) {
+            Logger.error(this, e.getMessage(),e);
+            return false;
+        }
+    }
+
+    @Override
+    public String getPostgresScript() {
+        return "CREATE TABLE IF NOT EXISTS experiment ("
+                + "     id  varchar(255) primary key,\n"
+                + "     page_id varchar(255) not null,\n"
+                + "     name varchar(255) not null,\n"
+                + "     description varchar(255) not null,\n"
+                + "     status varchar(255) not null,\n"
+                + "     traffic_proportion jsonb not null,\n"
+                + "     traffic_allocation float4 not null,\n"
+                + "     mod_date timestamptz not null,\n"
+                + "     scheduling jsonb,\n"
+                + "     archived bool not null,\n"
+                + "     creation_date timestamptz not null,\n"
+                + "     created_by varchar(255) not null,\n"
+                + "     last_modified_by varchar(255) not null "
+                + ")";
+    }
+
+    @Override
+    public String getMSSQLScript() {
+        return "create table experiment (\n"
+                + "    id  NVARCHAR(255) primary key,\n"
+                + "    page_id NVARCHAR(255) not null,\n"
+                + "    name NVARCHAR(255) not null,\n"
+                + "    description NVARCHAR(255) not null,\n"
+                + "    status NVARCHAR(255) not null,\n"
+                + "    traffic_proportion NVARCHAR(MAX) not null,\n"
+                + "    traffic_allocation float not null,\n"
+                + "    mod_date datetimeoffset(3) not null,\n"
+                + "    scheduling NVARCHAR(MAX),\n"
+                + "    archived tinyint not null,\n"
+                + "    creation_date datetimeoffset(3) not null,\n"
+                + "    created_by NVARCHAR(255) not null,\n"
+                + "    last_modified_by NVARCHAR(255) not null\n"
+                + "\n"
+                + ");";
```

Comment:
we can do this to check if it exists in mssql https://github.com/dotCMS/core/blob/367f14a6d7bff4cb8168fb34a38ee88dd0dcaa53/dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task222208CreateVariantTable.java#L20

Reply:
I am checking the table exists already in the forceRun method:
```  
final DotDatabaseMetaData databaseMetaData = new DotDatabaseMetaData();
return !databaseMetaData.tableExists(DbConnectionFactory.getConnection(), "experiment");